backend/config.go

```diff
@@ -96,6 +97,27 @@ func (c *GrafanaCfg) Equal(c2 *GrafanaCfg) bool {
 	return true
 }
 
+// ProxyHash returns a hash of the configured proxy client cert contents
+// for use in datasource instance caching. It stores the hash so it only
+// needs to be calculated once per datasource instance.
+func (c *GrafanaCfg) ProxyHash() string {
+	if c == nil {
+		return ""
+	}
+	contents := c.config[proxy.PluginSecureSocksProxyClientCertContents]
+	if contents == "" {
+		return ""
+	}
+	hash := c.config[proxy.PluginSecureSocksProxyClientCertContentsHash]
+	if hash == "" {
+		h := fnv.New32a()
```

Comment:
Switched to 32 bit hash - we certainly don't need 16 characters for this in the key, and even in the extremely unlikely event of a collision there's no harm done since we check the cached config before using the cached instance.